### PR TITLE
Replace string-based keyword references with `Keyword` enum in CodeGeneration

### LINKED ISSUE #2008: Remove string-based reference of keyword tokens in CodeGeneration
It would be a task for a follow-up PR to keep this PR easier to review but it would be nice to do the same thing you’re doing for `TokenSpec` also for `KeywordSpec`, so that the last element of this array could become ``.keyword(.`self`)``.

_Originally posted by @ahoppen in https://github.com/apple/swift-syntax/pull/2006#discussion_r1285078525_

### PR DESCRIPTION
**Pull request by Matejkob**
Follow-up PR to #2006 
Resolves #2008

### CODE REVIEW

File: CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift [hunk 1]
```diff
@@ -535,11 +535,11 @@ public let ATTRIBUTE_NODES: [Node] = [
...
             kind: .token(choices: [
               .token(.identifier),
-              .keyword(text: "private"),
-              .keyword(text: "fileprivate"),
-              .keyword(text: "internal"),
-              .keyword(text: "public"),
-              .keyword(text: "open"),
+              .keyword(.`private`),
+              .keyword(.`fileprivate`),
+              .keyword(.`internal`),
```
Comment (ahoppen) [inline CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift:540]: Do these even need to be backpacked here. IIRC the only case that needs to be backtracked on the call side is `init`.

Same in a couple more places

**Review by ahoppen — COMMENTED** [review]
Wohooo. More type safety 🤩

File: CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift (hunk 1 above)
Comment (Matejkob) [reply CodeGeneration/Sources/SyntaxSupport/AttributeNodes.swift]: Great point thank you for it! ❤️ Just pushed the corrected version.

**Review by ahoppen — APPROVED** [review]
Thanks for your continued work to clean up CodeGeneration @Matejkob.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by Matejkob** [comment]
Thanks for this and other code reviews. 

Sending much love! ❤️

**Comment by ahoppen** [comment]
@Matejkob Looks like there’s a merge conflict now. Could you rebase the PR?

**Comment by Matejkob** [comment]
@ahoppen I've just resolved the merge conflict.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**